### PR TITLE
Simplify configuration

### PR DESCRIPTION
**Pull request by salbertson**
This should help address the confusion around configuration.

Here's an overview of the changes:

* All languages are now enabled by default and configuration files for each language have a naming convention.
* Removed extra layer of configuration, `.hound.yml`.
* Disabled most preferences by default and allow individual teams to easily configure Hound using sample configuration files. This change should make the effective style guide more clear; you shouldn't have to search around RuboCop's' config trying to find what rules to override.
* Updated documentation.
* Simplified `StyleGuide` classes and made them more consistent.
* Moved responsibility of excluding files to individual `StyleGuide` classes.

The migration path will require communicating with current users about renaming their config files, if they have any.

screenshot 2015-02-05 08 41 13

https://github.com/thoughtbot/hound/issues/576

### CODE REVIEW

File: spec/models/style_checker_spec.rb [hunk 1]
```diff
@@ -1,306 +1,110 @@
...
+      file = stub_commit_file("good.rb", "code")
+      pull_request = stub_pull_request(pull_request_files: [file])
+      line = double("Line", changed?: true)
+      violation = Violation.new(line: line)
+      ruby_style_guide = double("StyleGuide::Ruby", violations_in_file: violation)
+      allow(StyleGuide::Ruby).to receive(:new).and_return(ruby_style_guide)
 
-        violations = StyleChecker.new(pull_request).violations
-        messages = violations.flat_map(&:messages)
+      violation_messages = StyleChecker.new(pull_request).violations
```
Comment (salbertson) [inline spec/models/style_checker_spec.rb]: Harsh

File: app/assets/stylesheets/_base.scss [hunk 2]
```diff
@@ -21,6 +21,7 @@ body {
 
   p {
     padding-bottom: 1em;
+    line-height: 1.8em;
```
Comment (gylaz) [inline app/assets/stylesheets/_base.scss]: Seems unrelated to this PR. :moyai:
